mpp/shmem.h4.in

```diff
@@ -99,6 +99,12 @@ include(shmem_c_func.h4)dnl
 #ifdef __cplusplus
 } /* extern "C" */
 
+#if __cplusplus >= 201402L
+#define SHMEM_CXX_ATTRIBUTE_DEPRECATED [[deprecated]]
+#else
+#define SHMEM_CXX_ATTRIBUTE_DEPRECATED
```

Comment:
Doesn't `__attribute__ ((deprecated))` work for pre-C++14?  If so, might be able to do:
`#define SHMEM_CXX_ATTRIBUTE_DEPRECATED SHMEM_C_ATTRIBUTE_DEPRECTATED`?

Reply:
Kinda, but not really..  🙁  I tried, but couldn't find a portable location for the attribute on a function definition (versus declaration).